### ISSUE TITLE
"bazel run"/Proj_runner.sh does not allow setting the SDK version for iOS applications

### ISSUE THREAD

**Issue by davidsantiago** [google/bazel#131]
When I try to run any iOS project through bazel, it fails because it is attempting to use SDK version 8.1, and I am trying to use version 8.3.

You can see this with, for example, the examples/objc/PrenotCalculator project:

```
bazel David$ bazel run examples/objc/PrenotCalculator --ios_sdk_version=8.3
INFO: Found 1 target...
Target //examples/objc:PrenotCalculator up-to-date:
  bazel-bin/examples/objc/PrenotCalculator_bin
  bazel-bin/examples/objc/PrenotCalculator.ipa
  bazel-bin/examples/objc/PrenotCalculator.xcodeproj/project.pbxproj
INFO: Elapsed time: 0.122s, Critical Path: 0.00s

INFO: Running command line: bazel-bin/examples/objc/PrenotCalculator_runner.sh
iossim: ERROR: Invalid SDK version: 8.1
Supported device/SDK combinations:
  -d 'iPhone 4s' -s '8.3'
  -d 'iPhone 5' -s '8.3'
  -d 'iPhone 5s' -s '8.3'
  -d 'iPhone 6 Plus' -s '8.3'
  -d 'iPhone 6' -s '8.3'
  -d 'iPad 2' -s '8.3'
  -d 'iPad Retina' -s '8.3'
  -d 'iPad Air' -s '8.3'
  -d 'Resizable iPhone' -s '8.3'
  -d 'Resizable iPad' -s '8.3'
ERROR: Non-zero return code '3' from command: Process exited with status 3.
```

The result is the same whether or not the --ios_sdk_version argument is passed. I also did not have any success changing the outcome by setting the SDK_VERSION/SIM_DEVICE shell variables to the correct version.

Looking at the PrenotCalculator_runner.sh script, it appears that the SDK version is hard-coded into the generated script. 

```
# TODO(bazel-team): Use 7.1 and iPhone 6 when available.
readonly SDK_VERSION='8.1'
readonly SIM_DEVICE='iPhone 6'
readonly APP_DIR=$(mktemp -d -t extracted_app)
```

It seems like somehow through the path that the code takes, it correctly builds with the requested SDK version, but then fails to replace that SDK version into the generated _runner.sh script. The original template for this script appears to have the replaceable variables in a comment, where they do get replaced, but not in the actual setting of the variables: https://github.com/google/bazel/blob/master/tools/objc/ios_runner.sh.mac_template#L24

```
# TODO(bazel-team): Use %sdk_version% and %sim_device% when available.
readonly SDK_VERSION='8.1'
readonly SIM_DEVICE='iPhone 6'
readonly APP_DIR=$(mktemp -d -t extracted_app)
```

So it seems like there are possibly two problems: The correct device and sdk version are not making it to the point where the template is processed, and then the template doesn't have those parts replaceable. I'd be glad to try to track this down, but I thought I'd ask because the commit that created the template just says "Description redacted," so I'm not sure if this is worth my working on at the moment.

**Comment by philwo** [comment]
I can confirm this. If I install the iOS 8.1 Simulator, at least it somehow works (but runs the app in the older simulator, not the one part of the SDK specified on the command-line).

**Comment by davidsantiago** [comment]
While we're on the topic, would it make sense to create some way to specify the SDK version in the BUILD file rules? I don't think I've issued a single successful iOS build command that didn't require me to correctly pass the --ios_sdk_version option. Perhaps you could specify the SDK version in the build rule, with the command line flag overriding, and default to a sensible value if neither location says anything about it. I imagine querying the system for the currently active SDK version would violate reproducible builds by introducing system state into a build decision.

**Comment by davidsantiago** [comment]
I went diving through the code in this area. First of all, I realized that there is actually a way to do mostly what I suggested just above, using .bazelrc to set default options for the run command. Though, it is a bit of a shame that only your global .bazelrc OR the workspace's gets read. It'd be nice to have a way to set global options that the project overrides in its own file, since some settings will probably depend on the project and some will be user-specific.

It also seems that the whole option system is just not really set up to have dynamic settings in the way I suggested, so it'd be a deeper and more involved change than I imagined. The reasons are basically explained here: https://github.com/google/bazel/blob/master/src/main/java/com/google/devtools/common/options/OptionsBase.java#L25 That appears to only accept compile-time constant defaults (via the Option annotation) or from the command line. 

I also discovered that the --ios_sdk_version option is not the one that determines which iOS simulator to use during a "bazel run" command. It should use the --ios_simulator_version option, instead. So once that option is added to the command line, all that needs to be done is to update the ios_runner template so that it actually makes it from the command line to the runner script. I submitted a patch for that here: https://bazel-review.googlesource.com/#/c/1150/
